### ISSUE TITLE
Zap receipt for paid user should be accepted

### ISSUE THREAD

**Issue by mutatrum** [scsibug/nostr-rs-relay#160]
If I zap someone which is a paying member of a nostr-rs-relay, my zapper cannot post the zap receipt (kind 9735) as my zapper is not a paying member. I then get a DM requesting me to pay for that relay.

For kind 9735, the payment check should be done on the pubkey in the p tag, not the pubkey of the note.

https://github.com/nostr-protocol/nips/blob/master/57.md

**Comment by thesimplekid** [comment]
> For kind 9735, the payment check should be done on the pubkey in the p tag, not the pubkey of the note.

While I get the idea here as you want to ensure that they see the zap event this would be a spam vector and a way to bypass the requirements to pay for the relay as anyone could but a pub key in the p tag.  I think its best to require the event publisher to pay as it is now, and if you want your zaps to be seen on that relay pay for the zappers pubkey to be allowed.

**Comment by mutatrum** [comment]
That's a reversal of responsibility, as I then have to pay for my zappers pubkey on all paid relays where maybe someone is residing that is being zapped by the zapper. IMO, wouldn't be part of a paid subscription on a relay would mean being able to get zap receipts?

**Comment by thesimplekid** [comment]
>  wouldn't be part of a paid subscription on a relay would mean being able to get zap receipts?

I do not think it is a reversal of responsibility, paying for a relay is paying to be able to post to that relay.  

> as I then have to pay for my zappers pubkey on all paid relays where maybe someone is residing that is being zapped by the zapper.

You do not have to pay for it if you do not want your zap receipts to be seen on those relays, It's likely that whomever you are zapping shares a few free relays or paid relays that you have paid for and can see the zap receipts on those relays. If you have no relays in common interacting in anyway will be impossible not just with zaps. There is certainly more work to be done around relay discoverability but for a simply paid relay i believe the model is pay to post as it currently is.

**Comment by mutatrum** [comment]
Ok, then this can be closed.

The only nitpick I have but I don't know where this falls into the whole flow: I get a new DM with an invoice every time the zapper tries to post on a paid relay. So for some relays I get multiple DMs a day, which comes across as a bit overzealous. Easiest is probably to mute that pubkey, but that'll cut off the relay operator completely.

Thank you anyways!

**Comment by thesimplekid** [comment]
> The only nitpick I have but I don't know where this falls into the whole flow: I get a new DM with an invoice every time the zapper tries to post on a paid relay.

The dms are off by default and the relay operator chooses to turn them on or off. Though I agree the multiple a day is a bit annoying, I believe it send a new one if the old invoice has expired.
